inst/@symfun/isequal.m

```diff
@@ -0,0 +1,87 @@
+%% Copyright (C) 2017 NVS Abhilash
+%%
+%% This file is part of OctSymPy.
+%%
+%% OctSymPy is free software; you can redistribute it and/or modify
+%% it under the terms of the GNU General Public License as published
+%% by the Free Software Foundation; either version 3 of the License,
+%% or (at your option) any later version.
+%%
+%% This software is distributed in the hope that it will be useful,
+%% but WITHOUT ANY WARRANTY; without even the implied warranty
+%% of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See
+%% the GNU General Public License for more details.
+%%
+%% You should have received a copy of the GNU General Public
+%% License along with this software; see the file COPYING.
+%% If not, see <http://www.gnu.org/licenses/>.
+
+%% -*- texinfo -*-
+%% @documentencoding UTF-8
+%% @defmethod  @@symfun isequal (@var{f}, @var{g})
+%% Test if contents of two or more arrays are equal.
+%%
+%% Example:
+%% @example
+%% @group
+%% syms x
+%% f(x) = x + 1;
+%% g(x) = x + 1;
+%% isequal(f, g)
+%%   @result{} 1
+%% @end group
+%% @end example
+%%
+%% @example
+%% @group
+%% syms x y
+%% f(x) = x + 1;
+%% g(x, y) = x + 1;
+%% isequal(f, g)
+%%   @result{} 0
+%% @end group
+%% @end example
+%%
+%% Note that this function is implemented to avoid issue #740.
+%% It just checks the equality for variables of the @@symfun, if not equal,
+%% returns false otherwise delegates further checking to @@sym/isequaln.
+%%
+%% @end defmethod
+
+function t = isequal(x, y, varargin)
+
+  if isa (x, 'symfun') && isa (y, 'symfun')
+    vars_x = x.vars;
+    vars_y = y.vars;
+
+    if length (vars_x) ~= length (vars_y)
+      t = false;
+    else
+      for i = 1:length(vars_x)
+        v_x = vars_x{i}.flat;
+        v_y = vars_y{i}.flat;
+
+        if ~strcmp (v_x, v_y)
+          t = false;
+          return;
+        end
+      end
+      t = isequaln(x, y, varargin{:});
```

Comment:
I guess this is supposed to call the superclass method?  Might be clearer to do `isequal (formula (x), formula (y))`.